src/main/java/com/dev/cinema/controller/MovieController.java

```diff
@@ -0,0 +1,47 @@
+package com.dev.cinema.controller;
+
+import com.dev.cinema.dto.MovieRequestDto;
+import com.dev.cinema.dto.MovieResponseDto;
+import com.dev.cinema.mapper.MovieMapper;
+import com.dev.cinema.model.Movie;
+import com.dev.cinema.service.MovieService;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/movies")
+public class MovieController {
+    private final MovieService movieService;
+    private final MovieMapper movieMapper;
+
+    public MovieController(MovieService movieService, MovieMapper movieMapper) {
+        this.movieService = movieService;
+        this.movieMapper = movieMapper;
+    }
+
+    @GetMapping("/inject")
+    public String inject() {
+        movieService.add(new Movie("Fast & Furious", "action"));
+        movieService.add(new Movie("Matrix", "Science fiction"));
+        movieService.add(new Movie("Fearless", "Martial arts"));
+        return "Data added successfully";
+    }
+
+    @PostMapping
+    public void add(@RequestBody MovieRequestDto movieDto) {
+        movieService.add(movieMapper.convertDtoToMovie(movieDto));
+    }
+
+    @RequestMapping(method = RequestMethod.GET)
```

Comment:
Why not GetMapping?

Reply:
just wanted to try it out either ways
done